collections/docs/README.md

```diff
@@ -0,0 +1,231 @@
+state comes from complex concrete golang objects (strings, ints, structs, etc.).
+
+Collections allows you to work with state as if they were normal golang objects and removes the need
+for you to think of your state as raw bytes in your code.
+
+It also allows you to migrate your existing state without causing any state breakage that forces you into
+tedious and complex chain state migrations.
+
+## Installation
+
+To install collections in your cosmos-sdk chain project, run the following command:
+
+```shell
+go get cosmossdk.io/collections
+```
+
+## Core types
+
+Collections offers 5 different APIs to work with state, which will be explored in the next sections, these APIs are:
+- ``Map``: to work with typed arbitrary KV pairings.
+- ``KeySet``: to work with just typed keys
+- ``Item``: to work with just one typed value
+- ``Sequence``: which is a monotonically increasing number.
+- ``IndexedMap``: which combines ``Map`` and `KeySet` to provide a `Map` with indexing capabilities.
+
+## Preliminary components
+
+Before exploring the different collections types and their capability it is necessary to introduce
+the three components that every collection shares. In fact when instantiating a collection type by doing, for example,
+```collections.NewMap/collections.NewItem/...``` you will find yourself having to pass them some common arguments.
+
+For example, in code:
+
+```go
+package collections
+
+import (
+    "cosmossdk.io/collections"
+    storetypes "cosmossdk.io/store/types"
+    sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var AllowListPrefix = collections.NewPrefix(0)
+
+type Keeper struct {
+	Schema    collections.Schema
+	AllowList collections.KeySet[string]
+}
+
+func NewKeeper(storeKey *storetypes.KVStoreKey) Keeper {
+	sb := collections.NewSchemaBuilder(sdk.OpenKVStore(storeKey))
+
+	return Keeper{
+		AllowList: collections.NewKeySet(sb, AllowListPrefix, "allow_list", collections.StringKey),
+	}
+}
+
+```
+
+Let's analyse the shared arguments, what they do, and why we need them.
+
+### SchemaBuilder
+
+The first argument passed is the ``SchemaBuilder``
+
+Schema builder is a structure that keeps track of all the state of a module, it is not required by the collections
+themselve to deal with state, but it offers a dynamic and reflective way for clients to explore a module's state.
+
+We instantiate a ``SchemaBuilder`` by passing it a function that given the modules store key returns the module's specific store.
+
+We then need to pass the schema builder to every collection type we instantiate in our keeper, in our case the `AllowList`.
+
+### Prefix
+
+The second argument passed to our ``KeySet`` is a `collections.Prefix`, a prefix represents a partition of the module's KVStore
+where all the state of a specific collection will be saved. Since a module can have multiple collections, example:
+
+- Module params will become a `collections.Item`,
+- The `AllowList` is a `collections.KeySet`.
+- etc.
+
```

Comment:
```suggestion
where all the state of a specific collection will be saved. 

Since a module can have multiple collections, the following is expected:
- module params will become a `collections.Item`
- the `AllowList` is a `collections.KeySet`

```